CRM/Logging/Schema.php

```diff
@@ -776,8 +780,16 @@ public function triggerInfo(&$info, $tableName = NULL, $force = FALSE) {
         $sqlStmt .= "NEW.$column, ";
         $deleteSQL .= "OLD.$column, ";
       }
-      $sqlStmt .= "CONNECTION_ID(), @civicrm_user_id, '{eventName}');";
-      $deleteSQL .= "CONNECTION_ID(), @civicrm_user_id, '{eventName}');";
+      if (civicrm_api3('Setting', 'getvalue', array('name' => 'logging_uniqueid_date', 'group' => 'core'))) {
+        $sqlStmt .= "@uniqueID, @civicrm_user_id, '{eventName}');";
+        $deleteSQL .= "@uniqueID, @civicrm_user_id, '{eventName}');";
+      }
+      else {
+        // The log tables have not yet been converted to have varchar(17) fields for log_conn_id.
+        // Continue to use the less reliable connection_id for al tables for now.
+        $sqlStmt .= "CONNECTION_ID(), @civicrm_user_id, '{eventName}');";
+        $deleteSQL .= "@uniqueID, @civicrm_user_id, '{eventName}');";
```

Comment:
@eileenmcnaughton , why does the `delete` trigger use different ID than the `create/update` triggers? Wouldn't this produce an error when deleting from a table whose is still based on `int`?

Aside: To actually produce this situation, I had to (1) load old code, (2) enable triggers, (3) load new code, (4) `drush cvapi system.flush triggers=1`, and then (5) delete a record. Easier to see when `delete` is really `delete` (not `Contact.delete`).